### PR TITLE
[SOAR-17259] Rapid7 Insight Agent

### PR DESCRIPTION
**Pull request by ablakley-r7**
### Description

Describe the proposed changes:

  - Insight Agent Search Agents is failing in cloud due to a timeout
  - Initial step, reduce logging of results to increase performance

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py [hunk 1]
```diff
@@ -399,8 +397,8 @@ def _get_next_page_of_agents(self, results_object: dict) -> (bool, dict, list):
         :param results_object: dict
         :return: tuple (boolean, dict (results), list (agents))
         """
-        self.logger.info("Getting next page of agents.")
         next_cursor = results_object.get("data").get("organization").get("assets").get("pageInfo").get("endCursor")
+        self.logger.info(f"Getting next page of agents using cursor {next_cursor}")
         payload = {
             "query": "query( $orgId:String! $nextCursor:String! ) { organization(id: $orgId) { assets( first: 10000, after: $nextCursor ) { pageInfo { hasNextPage endCursor } edges { node { id platform host { vendor version description hostNames { name } primaryAddress { ip mac } uniqueIdentity { source id } attributes { key value } } agent { agentSemanticVersion agentStatus quarantineState { currentState } } } } } } }",
```
Comment (joneill-r7) [inline plugins/rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py:403]: looking at some other actions can this query not be tailored to query the agent value? I think we know the agent type, so could do something like 

```
"query( $orgId:String! $nextCursor:String!  $agentValue:String!).....
"variables": {"orgId": self.org_key, "nextCursor": next_cursor, "agentValue": agent_input },
```

Just seems weird the query API can't take any input other than the org ID being requested!

**Review by igorski-r7 — APPROVED** [review]

File: plugins/rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py (hunk 1 above)
Comment (ablakley-r7) [reply plugins/rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py:403]: From what I've read and tried on the endpoint, it doesn't seem possible. The last recommended advice was to query for all agents then filter. Following up through the API docs now to see if there's any update.

Comment (ablakley-r7) [reply plugins/rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py:403]: No this is not supported.

Comment (joneill-r7) [reply plugins/rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py:403]: Annoying we can try to improve the query but worth giving less logging a go to see if it helps as that seems to be the change between versions that are running successfully for the customer.

**Review by joneill-r7 — APPROVED** [review]

**Pull request merged**